Move upgraders from python to cpp

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70593

Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

tugsbayasgalan committed:
Move upgraders from python to cpp

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

cccclai commented on torch/jit/operator_upgraders.py:
```diff
@@ -1,120 +0,0 @@
-    # type here.
-    @no_type_check
-    @torch.jit.script
-    def full_0_4(size: List[int], fill_value: Union[int, float], *,
```
Where are they moved to now?

tugsbayasgalan replied:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/operator_upgraders/upgraders_entry.cpp

gmagogsfm commented on torch/csrc/jit/operator_upgraders/upgraders_entry.cpp:
```diff
@@ -57,19 +58,39 @@ def full_out_0_4(size:List[int], fill_value:number, *, out:Tensor) -> Tensor:
+    populate_content.insert(std::make_pair(entry.first, upgrader_graph));
   }
-
   populate_upgraders_map(std::forward<UpgraderMap>(populate_content));
```
Why is the `forward` here necessary? It should only be needed for arguments or rvalues. here `populate_content` is neither. What do you think?

gmagogsfm reviewed (approved):
Just some nits.

tugsbayasgalan commented on torch/jit/generate_bytecode.py:
```diff
@@ -0,0 +1,29 @@
+def format_bytecode(table):
+    # given a nested tuples, convert them to nested list
+    def listify(content):
+        if not isinstance(content, tuple):
```
This won't trigger. I think i was being paranoid

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

tugsbayasgalan committed:
Update on "Move upgraders from python to cpp"


Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

cccclai commented on torch/jit/generate_bytecode.py:
```diff
@@ -0,0 +1,28 @@
+        formatted_table[identifier] = content
+    return formatted_table
+
+def generate_bytecode() -> List:
```
nit: can we rename it to something like generate_upgraders_bytecode?
